client/Bootstrap/Navigation/README.md

```diff
@@ -0,0 +1,271 @@
+
+## Example usage
+
+To follow in a later PR.
+
+_Note/To do_: Describe/define how path parameters, as well as helper functions, such as 'go back to the previous Strimzi ui page', 'go to this page by ID', or 'is this page available by ID', are passed to pages mounted by the router.
+
+## Route configuration processing
+
+As mentioned, this component takes in [configuration](../../Pages/README.md) which maps [`Panels`](../../Panels/README.md) to routes, along with other supporting metadata, such as a translatable page names, what backend services are required to show the page, and what authorization(s) a user may need to see it. This configuration is processed via the Navigation's model, and will result in an object which is then iterated on in the Navigation's view layers to generate the UI's routing/navigation. For example, given the following configuration:
+
+```
+{
+    Homepage: {
+        contentComponent: HomePageComponent,
+        contexts: [
+            {
+                id: 'Home',
+                path: '/homepage'
+                name: 'HOME_TRANSLATION_KEY', // when translated will map to the string 'Home'
+                feature_flag: 'PAGE.HOME',
+                order: 0,
+                icon: HomeIcon,
+                pageType: HOME,
+                properties: {},
+                requiresMinimum: {
+                    backendSupportFor: {},
+                    authorizationOf: {}
+                }
+            }
+        ]
+    },
+    ConsumerGroups: {
+        contentComponent: () => import('Panels/ConsumerGroups'), // async import
+        contexts: [
+            {
+                id: 'ConsumerGroups.Cluster',
+                path: '/consumergroups'
+                name: 'CONSUMER_GROUPS_TRANSLATION_KEY', // when translated will map to the string 'Consumer Groups'
+                feature_flag: 'PAGE.CONSUMER_GROUPS',
+                order: 2,
+                icon: ConsumerGroupIcon,
+                pageType: NORMAL,
+                properties: {
+                    mode: 'Cluster'
+                },
+                requiresMinimum: {
+                    backendSupportFor: {
+                        Group: { READ }
+                    },
+                    authorizationOf: {
+                        Group: { READ }
+                    }
+                }
+            },
+            {
+                id: 'ConsumerGroups.Topic',
+                path: '/topic/:name/consumergroups'
+                name: 'CONSUMER_GROUPS_TRANSLATION_KEY', // when translated will map to the string 'Consumer Groups'
+                feature_flag: 'PAGE.TOPIC.CONSUMER_GROUPS',
+                order: 0,
+                icon: undefined,
+                pageType: NORMAL,
+                properties: {
+                    mode: 'Topic'
+                },
+                requiresMinimum: {
+                    backendSupportFor: {
+                        Group: { READ }
```

Comment:
Where are these constants defined? And what about subscription support?

Reply:
Did you see https://github.com/matthew-chirgwin/strimzi-ui/blob/12-routing-nav/docs/Architecture.md#entity-model ? I do not mention where the common constants come from, so will add that, but the idea is that the 'abilities' on the entity could be anything, and that they are contextual, known and used/referenced by the code that needs them.